ui/packages/shared/components/src/DateTimeRangePicker/RelativeDatePicker/index.tsx

```diff
@@ -84,14 +86,110 @@ const parseInput = (input: string): {value: number; unit: string} | null => {
   return {value, unit: unitLong[unit]};
 };
 
+export const RelativeDatePickerForPanel = ({
+  onChange = () => null,
+  range,
+  hidePopoverMenu,
+}: {
+  onChange: (from: RelativeDate | AbsoluteDate, to: RelativeDate | AbsoluteDate) => void;
+  range: DateTimeRange;
+  hidePopoverMenu: () => void;
+}): JSX.Element => {
+  const dateFromInRelative = range.from as RelativeDate;
+  const dateToInRelative = range.to as RelativeDate;
+
+  const [from] = useState<AbsoluteDate>(
+    range.from.isRelative()
+      ? new AbsoluteDate(
+          getHistoricalDate({
+            unit: dateFromInRelative.unit,
+            value: dateFromInRelative.value,
+          })
+        )
+      : (range.from as AbsoluteDate)
+  );
+  const [to] = useState<AbsoluteDate>(
+    range.to.isRelative()
+      ? new AbsoluteDate(
+          getHistoricalDate({
+            unit: dateToInRelative.unit,
+            value: dateToInRelative.value,
+          })
+        )
+      : (range.to as AbsoluteDate)
+  );
+
+  const getRelativeTimeRangeBetweenDates = (
+    timeRange: number
+  ): {unit: UNIT_TYPE; value: number} => {
+    const roundToHundredth = (value: number): number => {
+      return Number(value.toFixed(2));
+    };
+
+    if (timeRange < 1000 * 60 * 60) {
+      const timeRangeToMinutes = timeRange / 1000 / 60;
+      return {unit: UNITS.MINUTE, value: roundToHundredth(timeRangeToMinutes)};
+    }
+    if (timeRange < 1000 * 60 * 60 * 24) {
+      const timeRangeToHours = timeRange / 1000 / 60 / 60;
+      return {unit: UNITS.HOUR, value: roundToHundredth(timeRangeToHours)};
+    }
+    const timeRangeToDays = timeRange / 1000 / 60 / 60 / 24;
+    return {unit: UNITS.DAY, value: roundToHundredth(timeRangeToDays)};
+  };
+
+  const {unit, value} = useMemo(
+    () => getRelativeTimeRangeBetweenDates(to.getTime().getTime() - from.getTime().getTime()),
+    [from, to]
+  );
+
+  // When the list of presets is shown in the popover panel, we use this effect here to ensure that the
+  // absolute date range is converted to a relative date range and we then use the `onChange` prop to
+  // update the range in the `RelativeDatePicker` component below.
+  useEffect(() => {
+    onChange(new RelativeDate(unit, value), new RelativeDate(unit, 0));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [unit, value]);
+
+  const presetRanges = [
+    {value: 15, unit: UNITS.MINUTE, text: 'Last 15 minutes'},
+    {value: 1, unit: UNITS.HOUR, text: 'Last 1 hour'},
+    {value: 3, unit: UNITS.HOUR, text: 'Last 3 hours'},
+    {value: 6, unit: UNITS.HOUR, text: 'Last 6 hours'},
+    {value: 12, unit: UNITS.HOUR, text: 'Last 12 hours'},
+    {value: 1, unit: UNITS.DAY, text: 'Last 1 day'},
+  ];
+
+  return (
+    <div className="flex flex-col gap-3 items-center text-sm p-4">
+      {presetRanges.map(({value, unit, text}) => (
+        <Button
```

Comment:
These buttons are rendered as `primary` variants, should we go with `neutral` as default and only set `primary` for the selected range? 

Having all the buttons as `primary` looks a bit too busy. 
<img width="331" alt="Screenshot 2024-06-19 at 10 19 23 AM" src="https://github.com/parca-dev/parca/assets/1283424/2c99cb40-ec85-44f0-9148-f5334eb92557">

Reply:
sgtm!